internal/kubernetes/nodefilters.go

```diff
@@ -17,28 +17,50 @@ and limitations under the License.
 package kubernetes
 
 import (
+	"fmt"
+	"log"
 	"strings"
 	"time"
 
 	core "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/antonmedv/expr"
 )
 
-// NewNodeLabelFilter returns a filter that returns true if the supplied object
-// is a node with all of the supplied labels.
-func NewNodeLabelFilter(labels map[string]string) func(o interface{}) bool {
+// NewNodeLabelFilter returns a filter that returns true if the supplied node satisfies the boolean expression
+func NewNodeLabelFilter(expressionStr *string) (func(o interface{}) bool, error) {
+
 	return func(o interface{}) bool {
+		//This feels wrong but this is how the previous behavior worked so I'm only keeping it to maintain compatibility.
+		if *expressionStr == "" {
+			return true
+		}
+
 		n, ok := o.(*core.Node)
 		if !ok {
 			return false
 		}
-		for k, v := range labels {
-			if value, ok := n.GetLabels()[k]; value != v || !ok {
-				return false
-			}
+
+		nodeLabels := n.GetLabels()
+
+		parameters := map[string]interface{}{
+			"metadata": map[string]map[string]string{
+				"labels": nodeLabels,
+			},
 		}
-		return true
-	}
+
+		expression, err := expr.Compile(*expressionStr, expr.Env(parameters))
+		if err != nil {
+			log.Fatalf("Could not compile the node label expression: %v", err)
```

Comment:
went ahead and compiled outside of the body... seems like I was planning to do that anyway since I changed the signature to return an err, but i must have gotten distracted :laughing: